### DISCUSSION TITLE
Can I position window automatically, just like layout?

### DISCUSSION THREAD

**Discussion by NoRuBal** [flamendless/Slab#157] [Q&A] [answered]
Hello! First, thanks for all who maintaining this awesome library. I discovered this library recently, and I'm bit surprised that how easy to use this library and how everything is well-documented. I even had some issue about relative path but it was fixed within a day. How awesome!

To get the point, I have a question. My question is if there is any way to set windows position automatically. I have a code like this:
```
Slab.BeginWindow('login', {Title = ":: Login", AllowMove = false, ShowMinimize = false})
		Slab.Text("")
		Slab.BeginLayout('layout_login', {AlignX = 'center'})
			Slab.Text("Username:")
			Slab.SameLine()
			if Slab.Input("login_username", {Text = login_username}) then
				login_username = Slab.GetInputText() 
			end
			Slab.Text("Password:")
			Slab.SameLine()
			if Slab.Input("login_password", {Text = login_password, IsPassword = true, PasswordChar = "*"}) then
				login_password = Slab.GetInputText()
			end
			if Slab.Button("Login") then
				print("Login as ".. title.username .. ":" .. title.password)
			end
			Slab.SameLine()
			if Slab.Button("Register") then
				title.changescreen("register")
			end
		Slab.EndLayout()
		Slab.Text("")
Slab.EndWindow()
```
It looks like this:
![1](https://github.com/flamendless/Slab/assets/3197094/bdbafdfc-8e4e-47ef-82d9-2a6e85140088)

I want to position my window at center of the screen. I think I can get window size by Slab.GetWindowSize(), calculate it with my screen size and adjst X/Y coords of the window. But I wonder if it's possible to set window position automatically, just like setting AlignX element of layout.

Thank you for reading this post, and have a nice day!

**Comment by flamendless** [comment, marked as answer]
Hello there! Thanks for the kind words about this awesome library!.

>  My first question is if there is any way to set windows position automatically.

Yes, you can use the `X` and `Y` parameter in the `BeginWindow`.

> I want to position my window at center of the screen

You can do this with something like:
```lua
local ww, wh = 640, 480 -- slab window's size
local w, h = love.windows.getDimensions()
local cx = w/2 - ww/2
local cy = h/2 - wh/2
Slab.BeginWindow('login', {Title = ":: Login", AllowMove = false, ShowMinimize = false, X = cx, Y = cy, W = ww, H = wh, AutoSizeWindow = false})
...
```

> But I wonder if it's possible to set window position automatically, just like setting AlignX element of layout.

If you mean the slab window's position, unfortunately there is nothing implemented for that because it is trivial to calculate and provide your own coordinate and size

**Reply by NoRuBal** [reply]
Thank you for a fast reply with example code! 👍
